test/input.spec.js

```diff
@@ -35,4 +37,150 @@ module.exports.addTests = function({testRunner, expect}) {
       }, input)).toBe('contents of the file');
     });
   });
+
+  describe('Page.interceptFileChooser', function() {
+    it('should intercept file chooser that is attached to DOM', async({page, server}) => {
+      await page.setContent(`<input type=file>`);
+      const [chooser] = await Promise.all([
+        page.interceptFileChooser(),
+        page.click('input'),
+      ]);
+      expect(chooser).toBeTruthy();
+    });
+    it('should intercept file chooser that is not attached to DOM', async({page, server}) => {
+      const [chooser] = await Promise.all([
+        page.interceptFileChooser(),
+        page.evaluate(() => {
+          const el = document.createElement('input');
+          el.type = 'file';
+          el.click();
+        }),
+      ]);
+      expect(chooser).toBeTruthy();
+    });
+    it('should respect timeout', async({page, server}) => {
+      let error = null;
+      await page.interceptFileChooser({timeout: 1}).catch(e => error =e);
+      expect(error).toBeInstanceOf(puppeteer.errors.TimeoutError);
+    });
+    it('should respect default timeout', async({page, server}) => {
+      let error = null;
+      page.setDefaultTimeout(1);
+      await page.interceptFileChooser().catch(e => error =e);
+      expect(error).toBeInstanceOf(puppeteer.errors.TimeoutError);
+    });
+    it('should work with no timeout', async({page, server}) => {
+      const [chooser] = await Promise.all([
+        page.interceptFileChooser({timeout: 0}),
+        page.evaluate(() => setTimeout(() => {
+          const el = document.createElement('input');
+          el.type = 'file';
+          el.click();
+        }, 50))
+      ]);
+      expect(chooser).toBeTruthy();
+    });
+  });
+
+  describe('FileChooser.accept', function() {
```

Comment:
Let's have a test with multiple interceptors waiting for a single file chooser.

Reply:
Done.